### PR TITLE
graph-builder: test cyclic metadata

### PR DESCRIPTION
**Pull request by steveej**
This test is expected to fail with the according to the Cincinnati specification and implementation, which make use of a directed acyclic graph.

### CODE REVIEW

File: graph-builder/tests/net/quay_io/mod.rs [hunk 1]
```diff
@@ -324,3 +328,36 @@ fn fetch_release_public_must_succeed_with_schemes_missing_http_https() {
     .map(|url| registry::Registry::try_from_str(url).unwrap())
     .for_each(test);
 }
+
+#[test]
+fn fetch_relesae_with_cyclic_metadata_fails() -> Fallible<()> {
```
Comment (lucab) [inline graph-builder/tests/net/quay_io/mod.rs]: typo: s/relesae/release/

File: graph-builder/tests/images/test-cyclic-public-manual-0.0.2/release-metadata [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"kind": "cincinnati-metadata-v0",
+	"schema": 1,
```
Comment (lucab) [inline graph-builder/tests/images/test-cyclic-public-manual-0.0.2/release-metadata]: This field does not exist according to docs and codebase.

**Review by lucab — COMMENTED** [review]
Just a couple of minor things, test logic is fine otherwise.

File: graph-builder/tests/net/quay_io/mod.rs (hunk 1 above)
Comment (steveej) [reply graph-builder/tests/net/quay_io/mod.rs]: _relesae_ would be a nice plural form though ;-)

**Comment by steveej** [comment]
@lucab all comments addressed

**Comment by steveej** [comment]
> @steveeJ: you cannot LGTM your own PR.

my bad. @lucab mind putting it back? :-)

**Pull request merged**